pkg/reconciler/common/common.go

```diff
@@ -77,3 +93,19 @@ func getTriggerRes(informer informer.TektonTriggerInformer) (*v1alpha1.TektonTri
 	res, err := informer.Lister().Get(v1alpha1.TriggerResourceName)
 	return res, err
 }
+
+func CheckUpgradePending(tc v1alpha1.TektonComponent) (bool, error) {
+	labels := tc.GetLabels()
+	ver, ok := labels[tektoninstallerset.ReleaseVersionKey]
+	if !ok {
+		return true, nil
+	}
+	operatorVersion, err := OperatorVersion(context.TODO())
```

Comment:
nit, for the future: we should probably "chain" `context` 👼🏼

Reply:
absolutely, :+1: